rosbridge_library/src/rosbridge_library/internal/services.py

```diff
@@ -141,20 +152,29 @@ def call_service(
         service_class, service, callback_group=ReentrantCallbackGroup()
     )
 
-    if not client.wait_for_service(server_timeout_time):
+    if not client.wait_for_service(server_ready_timeout):
         node_handle.destroy_client(client)
         raise InvalidServiceException(service)
 
     future = client.call_async(inst)
-    while rclpy.ok() and not future.done():
+    start_time = time.monotonic()
+    while (
+        rclpy.ok() and not future.done() and time.monotonic() - start_time < server_response_timeout
```

Comment:
Is there no option to never time out? Like a -1 value or something?

Reply:
We could treat any non-positive value as "no timeout"